hooks/hooks.go

```diff
@@ -1,50 +1,167 @@
+	"github.com/xmidt-org/argus/chrysom"
+	"github.com/xmidt-org/argus/model"
+	"github.com/xmidt-org/bascule"
 	"github.com/xmidt-org/webpa-common/webhook"
-	"github.com/xmidt-org/webpa-common/xmetrics"
+	"io/ioutil"
+	"net/http"
 )
 
 // Options describes the parameters needed to configure the webhook endpoints
 type Options struct {
-
-	//APIRouter is assumed to be a subrouter with the API prefix path (i.e. 'api/v2')
+	// APIRouter is assumed to be a subrouter with the API prefix path (i.e. 'api/v2')
 	APIRouter *mux.Router
 
-	//RootRouter is the main Tr1d1um router
-	RootRouter *mux.Router //Router with empty path prefix
-
 	Authenticate *alice.Chain
 
-	SoAProvider string
-
-	M xmetrics.Registry
-
-	Host         string
-	HooksFactory *webhook.Factory
-	Log          kitlog.Logger
-	Scheme       string
+	Log                kitlog.Logger
+	WebhookStoreConfig chrysom.ClientConfig
 }
 
 // ConfigHandler configures a given handler with webhook endpoints
 func ConfigHandler(o *Options) {
-	hooksRegistry, hooksHandler := o.HooksFactory.NewRegistryAndHandler(o.M)
+	r, _ := NewRegistry(RegistryConfig{
+		Logger:   o.Log,
+		Listener: nil,
+		Config:   o.WebhookStoreConfig,
+	})
+
+	o.APIRouter.Handle("/hook", o.Authenticate.ThenFunc(r.UpdateRegistry)).Methods(http.MethodPost)
+	o.APIRouter.Handle("/hooks", o.Authenticate.ThenFunc(r.GetRegistry)).Methods(http.MethodGet)
 
-	o.APIRouter.Handle("/hook", o.Authenticate.ThenFunc(hooksRegistry.UpdateRegistry)).Methods(http.MethodPost)
-	o.APIRouter.Handle("/hooks", o.Authenticate.ThenFunc(hooksRegistry.GetRegistry)).Methods(http.MethodGet)
+}
+
+type HookStore interface {
+	chrysom.Pusher
+	chrysom.Reader
+}
 
-	selfURL := &url.URL{
-		Scheme: o.Scheme,
-		Host:   o.Host,
+type Registry struct {
+	hookStore HookStore
+	config    RegistryConfig
+}
+
+type RegistryConfig struct {
+	Logger   kitlog.Logger
+	Listener chrysom.ListenerFunc
+	Config   chrysom.ClientConfig
+}
+
+func NewRegistry(config RegistryConfig) (*Registry, error) {
+	argus, err := chrysom.CreateClient(config.Config, chrysom.WithLogger(config.Logger))
+	if err != nil {
+		return nil, err
+	}
+	if config.Listener != nil {
+		argus.SetListener(config.Listener)
 	}
+	return &Registry{
+		config:    config,
+		hookStore: argus,
+	}, nil
+}
 
-	//Initialize must take the router without any prefixes
-	o.HooksFactory.Initialize(o.RootRouter, selfURL, o.SoAProvider, hooksHandler, o.Log, o.M, time.Now)
+// jsonResponse is an internal convenience function to write a json response
+func jsonResponse(rw http.ResponseWriter, code int, msg string) {
+	rw.Header().Set("Content-Type", "application/json")
+	rw.WriteHeader(code)
+	type responseMessage struct {
+		msg string `json:"message"`
+	}
+	data, _ := json.Marshal(&responseMessage{msg: msg})
+	rw.Write(data)
+}
+
+// update is an api call to processes a listener registration for adding and updating
+func (r *Registry) GetRegistry(rw http.ResponseWriter, req *http.Request) {
+	owner := ""
```

Comment:
Is owner something we expect to fill out as a later PR?

Reply:
`owner` is updated from the auth.
```golang
owner = auth.Token.Principal()
```